src/sql/workbench/services/objectExplorer/browser/iconRenderer.ts

```diff
@@ -22,22 +23,32 @@ class IconRenderer {
 		return iconUid;
 	}
 
-	public getIconUid(path: URI | IconPath): string | undefined {
+	public getIconUid(path: IconPath): string | undefined {
 		if (!path) { return undefined; }
-		let iconPath: IconPath = this.toIconPath(path);
+		const iconPath: ThemedIconUri = this.toIconPath(path);
 		return `icon${hash(iconPath.light.toString() + iconPath.dark.toString())}`;
 	}
 
-	private toIconPath(path: URI | IconPath): IconPath {
-		if (URI.isUri(path)) {
-			let singlePath = path;
-			return { light: singlePath, dark: singlePath };
+	private toIconPath(path: IconPath): ThemedIconUri {
+		let light, dark: string | URI;
+
+		if (URI.isUri(path) || (typeof (path) === 'string')) {
+			light = dark = path;
 		} else {
-			return path;
+			light = path.light;
+			dark = path.dark;
 		}
+		return {
+			light: this.toUri(light),
+			dark: this.toUri(dark)
+		};
+	}
+
+	private toUri(path: string | URI): URI {
```

Comment:
Couple of these functions could be moved out of the class - no reason for them to be in here.

Reply:
these methods are local helper methods marked as private, I don't think it makes sense to move them to anywhere else. I'd leave these here.